src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.ts

```diff
@@ -0,0 +1,164 @@
+    });
+  }
+
+  setProfile(profile: any) {
+    const customProfile = _.find(this.profiles, (p) => {
+      return p.name === 'custom';
+    });
+
+    if (profile.name === 'custom') {
+      if (!customProfile) {
+        this.profiles.push(profile);
+      }
+    } else {
+      if (customProfile) {
+        this.profiles.splice(this.profiles.indexOf(customProfile), 1);
+      }
+    }
+
+    this.osdRecvSpeedForm.controls.profile.setValue(profile.name);
+    Object.entries(profile.values).forEach(([name, value]) => {
+      this.osdRecvSpeedForm.controls[name].setValue(value);
+    });
+  }
+
+  onCustomizeProfileChange() {
+    if (this.osdRecvSpeedForm.getValue('customizeProfile')) {
+      const values = {};
+      this.profileAttrs.forEach((attr) => {
+        values[attr] = this.osdRecvSpeedForm.getValue(attr);
+      });
+      const customProfile = {
+        name: 'custom',
+        text: 'Custom',
+        values: values
+      };
+      this.setProfile(customProfile);
+    } else {
+      this.setProfile(this.profiles[0]);
+    }
+  }
+
+  getStoredProfile(callbackFn: Function) {
+    const observables = [];
+    this.profileAttrs.forEach((configName) => {
+      observables.push(this.configService.get(configName));
+    });
+
+    observableForkJoin(observables)
+      .pipe(
+        mergeMap((configOptions) => {
+          const result = {};
+          configOptions.forEach((configOption) => {
+            if ('value' in configOption) {
+              configOption.value.forEach((value) => {
+                if (value['section'] === 'osd') {
+                  result[configOption.name] = Number(value.value);
+                }
+              });
+            }
+          });
+          return of(result);
+        })
+      )
+      .subscribe((resp) => {
+        const profile = _.find(this.profiles, (p) => {
+          return _.isEqual(p.values, resp);
+        });
+
+        this.osdRecvSpeedForm.controls.customizeProfile.setValue(false);
+
+        if (profile) {
+          return callbackFn(profile);
+        }
+
+        if (Object.entries(resp).length === 4) {
+          this.osdRecvSpeedForm.controls.customizeProfile.setValue(true);
+          return callbackFn(Object({ name: 'custom', text: 'Custom', values: resp }));
+        }
+
+        return callbackFn(this.profiles[0]);
+      });
+  }
+
+  onProfileChange(selectedProfileName) {
+    const selectedProfile =
+      _.find(this.profiles, (p) => {
+        return p.name === selectedProfileName;
+      }) || this.profiles[0];
+
+    this.setProfile(selectedProfile);
+  }
+
+  submitAction() {
+    const options = {};
+    this.profileAttrs.forEach((attr) => {
+      options[attr] = { section: 'osd', value: this.osdRecvSpeedForm.getValue(attr) };
+    });
+
+    this.configService.bulkCreate({ options: options }).subscribe(
+      () => {
+        this.notificationService.show(
+          NotificationType.success,
+          'OSD recovery speed profile "' +
```

Comment:
Please add translation.

Reply:
You mean I should add a translation via transifex?